frontend/catalyst/device/qjit_device.py

```diff
@@ -255,6 +255,16 @@ def get_qjit_device_capabilities(target_capabilities: DeviceCapabilities) -> Set
             }
         )
 
+    # Optionally enable runtime-powered StatePrep quantum gate
+    if any(ng.invertible for ng in target_capabilities.native_ops.values()):
+        qjit_capabilities.native_ops.update(
+            {
+                "StatePrep": OperationProperties(
+                    invertible=True, controllable=True, differentiable=True
+                )
+            }
+        )
```

Comment:
This doesn't look right. The gate needs to be added to an actual device TOML file, although in the meantime this can done for testing I suppose. But I don't see a reason for the "is any gate convertible" check.

Reply:
I have added the gate to the TOML file of lightning qubit. However, when the program reaches `bind_native_operation` the gate arrives decomposed. The above was added to prevent such decomposition. How do I make sure the TOML file will prevent the decomposition?

Reply:
Removed. Done.

Reply:
Yes I think for now having it here until the TOML pr is merged and we updated the version of lightning we build against is fine. It's just the if statement doesn't make sense, and we need to remember to remove it before merging :)